Packs/PAN-OS/Integrations/Panorama/Panorama.py

```diff
@@ -14054,6 +14090,381 @@ def pan_os_get_audit_comment_command(args: dict) -> CommandResults:
     )
 
 
+def get_all_profile_names_from_profile_type(profile_type: str, device_group: str) -> list:
+    """
+    Retrieves all profile names from a specified profile type.
+
+    Args:
+        profile_type: The type of profile to retrieve, 'vulnerability' or 'spyware'.
+        device_group: If device_group was provided as a command argument, it will override the configured one.
+
+    Returns:
+        A list of profile names associated with the specified profile type.
+    """
+    # if device_group was provided as a command argument, it will override the configured one
+    if device_group:
+        if device_group == "shared":
+            xpath = "/config/shared/"
+        else:
+            xpath = (
+                "/config/devices/entry/device-group/entry[@name='"
+                + device_group
+                + "']/"
+            )
+    else:
+        xpath = "/config/devices/entry/vsys/entry[@name='" + VSYS + "']/"
+    xpath += f"profiles/{profile_type}"
+
+    raw_response = get_security_profile(xpath)
+    profiles = raw_response["response"]["result"][f"{profile_type}"]["entry"]
```

Comment:
What if the keys aren't there?

Reply:
The keys are from the api so if they wont be there the api changed so i think it doesnt matter. Let me know if there is another solution :)

Reply:
Use .get() instead. 
raw_response.get('response', {}].get()...